contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,454 @@
+    */
+    function redeemNativeToken(bytes32 _proposalId) public returns(uint256 amount) {
+
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal rewards to zero to prevent reentrancy attack.
+        proposal.nativeTokenReward = 0;
+
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.nativeTokenReward != 0) {
+                proposal.nativeTokenRewardLeft = _proposal.nativeTokenReward;
+            }
+        }
+        amount = _proposal.nativeTokenReward;
+        if (amount > 0) {
+            require(Controller(avatar.owner()).mintTokens(amount, _proposal.beneficiary, address(avatar)));
+            emit RedeemNativeToken(address(avatar), _proposalId, _proposal.beneficiary, amount);
+        }
+    }
+
+    /**
+    * @dev RedeemEther reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return amount ether redeemed amount
+    */
+    function redeemEther(bytes32 _proposalId) public returns(uint256 amount) {
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal rewards to zero to prevent reentrancy attack.
+        proposal.ethReward = 0;
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.ethReward != 0) {
+                proposal.ethRewardLeft = _proposal.ethReward;
+            }
+        }
+        amount = _proposal.ethReward;
+
+        if (amount > 0) {
+            require(Controller(avatar.owner()).sendEther(amount, _proposal.beneficiary, avatar));
+            emit RedeemEther(address(avatar), _proposalId, _proposal.beneficiary, amount);
+        }
+    }
+
+    /**
+    * @dev RedeemNativeToken reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return amount the external token redeemed amount
+    */
+    function redeemExternalToken(bytes32 _proposalId) public returns(uint256 amount) {
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal rewards to zero to prevent reentrancy attack.
+        proposal.externalTokenReward = 0;
+
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.externalTokenReward != 0) {
+                proposal.externalTokenRewardLeft = _proposal.externalTokenReward;
+            }
+        }
+
+        if (proposal.externalToken != IERC20(0) && _proposal.externalTokenReward > 0) {
+            amount = _proposal.externalTokenReward;
+            if (amount > 0) {
```

Comment:
This `if` is duplicated, as it checks the same think as the `if` above, please remove.